src/Bicep.LangServer/Handlers/BicepDeploymentParametersHandler.cs

```diff
@@ -148,7 +148,7 @@ private bool IsExpression(SyntaxBase modifier)
             return modifier is ParameterDefaultValueSyntax parameterDefaultValueSyntax &&
                 parameterDefaultValueSyntax.DefaultValue is ExpressionSyntax expressionSyntax &&
                 expressionSyntax is not null &&
-                expressionSyntax is not StringSyntax &&
+                expressionSyntax is not StringSyntax &&  // This line is what is really causing the issue. Strings that contain ${} should be treated as expressions.
```

Comment:
... and this is where the fix should be, not in the caller

(remove the comment)

Reply:
```suggestion
                expressionSyntax is not StringSyntax &&
```

Reply:
The code you have wouldn't take into account escape sequences.  Use StringSyntax.isInterpolated(), example:
```
 public override void VisitImportDeclarationSyntax(ImportDeclarationSyntax syntax)
        {
            base.VisitImportDeclarationSyntax(syntax);

            TypeSymbol declaredType;
            if (!features.ExtensibilityEnabled)
            {
                declaredType = ErrorType.Create(DiagnosticBuilder.ForPosition(syntax).ImportsAreDisabled());
            }
            else if (syntax.SpecificationString is StringSyntax specificationString && specificationString.IsInterpolated())
            {
        ``

Reply:
```suggestion
                expressionSyntax is not StringSyntax &&
                specificationString.IsInterpolated() &&
```